File: backend/src/main/java/com/contractly/signature/model/SignatureRecord.java
package com.contractly.signature.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SignatureRecord {
    private Long id;
    private Long contractId;
    private String signerName;
    private String signerEmail;
    private String signatureData;   // base64 encoded signature image
    private String ipAddress;
    private String userAgent;
    private String documentHash;
    private LocalDateTime signedAt;
}

File: backend/src/models/SignatureRecord.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface ISignatureRecord extends Document {
  contractId: mongoose.Types.ObjectId;
  signerName: string;
  signerEmail: string;
  signatureData: string; // base64
  ipAddress?: string;
  userAgent?: string;
  documentHash?: string;
  signedAt: Date;
}

const SignatureRecordSchema: Schema = new Schema(
  {
    contractId: { type: Schema.Types.ObjectId, ref: 'Contract', required: true },
    signerName: { type: String, required: true },
    signerEmail: { type: String, required: true },
    signatureData: { type: String, required: true },
    ipAddress: { type: String },
    userAgent: { type: String },
    documentHash: { type: String },
    signedAt: { type: Date, default: Date.now },
  }
);

export default mongoose.model<ISignatureRecord>('SignatureRecord', SignatureRecordSchema);
